plutus-core/index-envs/src/Data/DeBruijnEnv.hs

```diff
@@ -33,22 +38,23 @@ instance DeBruijnEnv (BRAL.RAList a) where
     {-# INLINABLE cons #-}
     cons = BRAL.Cons
     {-# INLINABLE index #-}
-    index = BRAL.safeIndex
+    index e i = BRAL.safeIndexZero e (i-1)
     {-# INLINABLE unsafeIndex #-}
-    unsafeIndex = BRAL.index
+    unsafeIndex e i = BRAL.unsafeIndexZero e (i-1)
 
 -- | A sequence implemented by a map from "levels" to values and a counter giving the "current" level.
-data RelativizedMap a = RelativizedMap (IM.IntMap a) {-# UNPACK #-} !Int
+data RelativizedMap a = RelativizedMap (IM.IntMap a) {-# UNPACK #-} !Word64
+    deriving Show
 
 instance DeBruijnEnv (RelativizedMap a) where
     type Element (RelativizedMap a) = a
 
     {-# INLINABLE empty #-}
     empty = RelativizedMap mempty 0
     {-# INLINABLE cons #-}
-    cons a (RelativizedMap im l) = RelativizedMap (IM.insert l a im) (l+1)
+    cons a (RelativizedMap im l) = RelativizedMap (IM.insert (fromIntegral l) a im) (l+1)
     {-# INLINABLE index #-}
-    index (RelativizedMap im l) w = IM.lookup (l - fromIntegral w) im
+    index (RelativizedMap im l) w = IM.lookup (fromIntegral l - fromIntegral w) im
```

Comment:
Is this missing a change to make it 1-indexed?

Reply:
relativizedmap is 1-indexed